### PR TITLE
Add Domain namespace

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- Enclose everything in Domain in a `domain` namespace
- DomainCreators -> `domain::creators`
- CoordinateMaps -> `domain::CoordinateMaps`

### Types of changes:

- [ ] Bugfix
- [x] New feature

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by nilsvu — COMMENTED** [review]
👍 good change, comments on namespace naming guidelines welcome in #727

File: src/Domain/ElementId.cpp [hunk 1]
```diff
@@ -67,45 +69,45 @@ void ElementId<VolumeDim>::pup(PUP::er& p) noexcept {
 }
 
 template <size_t VolumeDim>
-std::ostream& operator<<(std::ostream& os, const ElementId<VolumeDim>& id) {
-  os << "[B" << id.block_id() << ',' << id.segment_ids() << ']';
+std::ostream& operator<<(std::ostream& os,
+                         const domain::ElementId<VolumeDim>& id) {
+  ::operator<<(os << "[B" << id.block_id() << ',', id.segment_ids()) << ']';
```
Comment (kidder) [inline src/Domain/ElementId.cpp]: what is going on here? why do I need to scope the first operator?

File: src/Domain/ElementIndex.cpp [hunk 2]
```diff
@@ -28,12 +30,15 @@ SegmentIndex::SegmentIndex(size_t block_id,
 static_assert(std::is_pod<SegmentIndex>::value, "SegmentIndex is not POD");
 static_assert(sizeof(SegmentIndex) == sizeof(int),
               "SegmentIndex does not fit in an int");
+}  // namespace domain
 
-std::ostream& operator<<(std::ostream& s, const SegmentIndex& index) noexcept {
+std::ostream& operator<<(std::ostream& s,
```
Comment (kidder) [inline src/Domain/ElementIndex.cpp]: why are some stream operators in the namespace and some not?

File: src/Domain/ElementMap.hpp [hunk 3]
```diff
@@ -43,10 +46,10 @@ class ElementMap {
   ElementMap() = default;
   /// \endcond
 
-  ElementMap(
-      ElementId<Dim> element_id,
-      std::unique_ptr<CoordinateMapBase<Frame::Logical, TargetFrame, Dim>>
-          block_map) noexcept;
+  ElementMap(ElementId<Dim> element_id,
```
Comment (kidder) [inline src/Domain/ElementMap.hpp]: you don't appear to be consistent in putting `domain::` in front of namespace members within the namespace

**Comment by nilsdeppe** [comment]
- Fixed IWYU
- Fixed GCC compilation

**Comment by wthrowe** [comment]
Please split the unrelated changes out of the main commit.

**Comment by nilsdeppe** [comment]
Umm... It is already...?? These are the exact changes necessary to introduce the namespace, nothing more.

File: src/Domain/Neighbors.cpp [hunk 4]
```diff
@@ -47,19 +44,32 @@ void Neighbors<VolumeDim>::pup(PUP::er& p) noexcept {
   p | orientation_;
 }
 
+template <size_t VolumeDim>
+std::ostream& operator<<(std::ostream& os,
+                         const domain::Neighbors<VolumeDim>& n) {
+  ::operator<<(os << "Ids = ", n.ids())
+      << "; orientation = " << n.orientation();
+  // os << std::string("Ids = ") << n.ids()
```
Comment (nilsdeppe) [inline src/Domain/Neighbors.cpp]: Note to me: delete the commented code

File: tests/Unit/Pypp/PyppFundamentals.hpp [hunk 5]
```diff
@@ -221,6 +221,7 @@ struct FromPyObject<long, std::nullptr_t> {
 #if PY_MAJOR_VERSION == 2 && PY_MINOR_VERSION == 7
     } else if (not PyInt_Check(t) and not PyLong_Check(t)) {
 #elif PY_MAJOR_VERSION == 3
+      // NOLINTNEXTLINE(hicpp-signed-bitwise)
```
Comment (nilsdeppe) [inline tests/Unit/Pypp/PyppFundamentals.hpp:224]: Note to self: this can be factored.

File: tests/Unit/Domain/Test_BlockNeighbor.cpp [hunk 6]
```diff
@@ -13,23 +13,26 @@
 #include "Domain/SegmentId.hpp"  // IWYU pragma: keep
 #include "tests/Unit/TestHelpers.hpp"
 
+// IWYU pragma: no_include <type_traits>
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Test_BlockNeighbor.cpp]: Note to self: factor this line

File: tests/Unit/Domain/Test_Element.cpp [hunk 7]
```diff
@@ -18,48 +18,51 @@
...
-  const Element<VolumeDim> element_diff_neighbors(
-      id, typename Element<VolumeDim>::Neighbors_t{
-        {Direction<VolumeDim>::lower_xi(), neighbors}});
+  const domain::Element<VolumeDim> element_diff_neighbors(
+      id, typename domain::Element<VolumeDim>::Neighbors_t{
+              {domain::Direction<VolumeDim>::lower_xi(), neighbors}});
   CHECK(element != element_diff_neighbors);
   CHECK_FALSE(element == element_diff_neighbors);
 
-  CHECK(get_output(element) ==
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Test_Element.cpp]: Note to self: undo this, it's only formatting

File: src/Domain/Block.cpp [hunk 8]
```diff
@@ -38,75 +40,51 @@ void Block<VolumeDim, TargetFrame>::pup(PUP::er& p) {
   p | external_boundaries_;
 }
 
+template <size_t VolumeDim, typename TargetFrame>
```
Comment (nilsdeppe) [inline src/Domain/Block.cpp]: Note to self: don't reverse order here

File: src/Domain/Direction.cpp [hunk 9]
```diff
@@ -62,13 +62,16 @@ std::ostream& operator<<(std::ostream& os,
   os << direction.dimension();
   return os;
 }
+}  // namespace domain
 
-template std::ostream& operator<<(std::ostream&, const Direction<1>&);
-template std::ostream& operator<<(std::ostream&, const Direction<2>&);
-template std::ostream& operator<<(std::ostream&, const Direction<3>&);
+template void domain::Direction<1>::pup(PUP::er&) noexcept;
```
Comment (nilsdeppe) [inline src/Domain/Direction.cpp]: Note to self: keep inside cond

**Review by nilsdeppe — COMMENTED** [review]
I've flagged a few things I can factor out.

The larger movement in some files, like the stream operator calls being explicit is because of ADL so are directly related to the domain namespace addition. Before ADL look in the global namespace, now it looks in domain. Since the STL stream operators are not in std or domain, it fails to find them. The solution is specifying the namespace explicitly.

File: src/Domain/Direction.cpp (hunk 9 above)
Comment (nilsdeppe) [reply src/Domain/Direction.cpp]: It is... I'm just dumb.

File: src/Domain/BlockNeighbor.cpp [hunk 10]
```diff
@@ -3,6 +3,8 @@
 
 #include "Domain/BlockNeighbor.hpp"
 
+#include "Utilities/GenerateInstantiations.hpp"
```
Comment (wthrowe) [inline src/Domain/BlockNeighbor.cpp]: Should be after system headers.

File: src/Domain/DomainHelpers.cpp [hunk 11]
```diff
@@ -54,23 +54,23 @@ std::vector<size_t> get_common_global_corners(
 // in block2, this function first repackages them into arrays and
 // then makes a std::pair of the arrays.
 template <size_t VolumeDim>
-std::pair<std::array<Direction<VolumeDim>, VolumeDim>,
-          std::array<Direction<VolumeDim>, VolumeDim>>
+std::pair<std::array<domain::Direction<VolumeDim>, VolumeDim>,
```
Comment (wthrowe) [inline src/Domain/DomainHelpers.cpp]: There seems to be a lot of unnecessary change going on in this file.  I think you only need to change four lines:
```patch
diff --git a/src/Domain/DomainHelpers.cpp b/src/Domain/DomainHelpers.cpp
index 56571316a..2b2800cd5 100644
--- a/src/Domain/DomainHelpers.cpp
+++ b/src/Domain/DomainHelpers.cpp
@@ -31,6 +31,7 @@
 #include "Utilities/GenerateInstantiations.hpp"
 #include "Utilities/Gsl.hpp"
 
+namespace domain {
 namespace {
 
 // returns the common corners in the global corner numbering scheme.
@@ -405,9 +406,9 @@ void set_identified_boundaries(
     const auto& face2 = pair.second;
     ASSERT(face1.size() == face2.size(),
            "Each set must have the same number of corners.");
-    size_t id1 = ::find_block_id_of_external_face<VolumeDim>(
+    size_t id1 = find_block_id_of_external_face<VolumeDim>(
         face1, corners_of_all_blocks);
-    size_t id2 = ::find_block_id_of_external_face<VolumeDim>(
+    size_t id2 = find_block_id_of_external_face<VolumeDim>(
         face2, corners_of_all_blocks);
     const auto face1_canon =
         get_common_local_corners<VolumeDim>(corners_of_all_blocks[id1], face1);
@@ -1055,4 +1056,5 @@ GENERATE_INSTANTIATIONS(INSTANTIATE, (1, 2, 3), (Frame::Grid, Frame::Inertial))
 #undef DIM
 #undef DTYPE
 #undef INSTANTIATE
+}  // namespace domain
 /// \endcond
```

File: src/Domain/ElementIndex.cpp [hunk 12]
```diff
@@ -78,52 +85,39 @@ size_t hash_value(const ElementIndex<VolumeDim>& index) noexcept {
...
+      const domain::ElementIndex<DIM(data)>& a,                           \
+      const domain::ElementIndex<DIM(data)>& b) noexcept;                 \
+  template size_t hash_value(                                             \
+      const domain::ElementIndex<DIM(data)>& index) noexcept;             \
+  template size_t std::hash<domain::ElementIndex<DIM(data)>>::operator()( \
+      const domain::ElementIndex<DIM(data)>& x) const noexcept;           \
+  template std::ostream& operator<<(                                      \
+      std::ostream& s, const domain::ElementIndex<DIM(data)>& index) noexcept;
+
+GENERATE_INSTANTIATIONS(INSTANTIATE, (1, 2, 3))
```
Comment (wthrowe) [inline src/Domain/ElementIndex.cpp]: Factor this out.

File: src/Domain/ElementIndex.hpp [hunk 13]
```diff
@@ -70,17 +76,18 @@ bool operator==(const ElementIndex<VolumeDim>& a,
 template <size_t VolumeDim>
 bool operator!=(const ElementIndex<VolumeDim>& a,
                 const ElementIndex<VolumeDim>& b) noexcept;
+}  // namespace domain
 
 template <size_t VolumeDim>
-size_t hash_value(const ElementIndex<VolumeDim>& index) noexcept;
+size_t hash_value(const domain::ElementIndex<VolumeDim>& index) noexcept;
```
Comment (wthrowe) [inline src/Domain/ElementIndex.hpp]: Should be in namespace domain for ADL.  I've probably missed some other cases of this.

File: src/Domain/ElementIndex.hpp [hunk 14]
```diff
@@ -43,10 +46,13 @@ class SegmentIndex {
   unsigned refinement_level_ : ElementIndex_detail::refinement_bits;
   unsigned index_ : ElementIndex_detail::max_refinement_level;
 };
+}  // namespace domain
 
 template <size_t VolumeDim>
-std::ostream& operator<<(std::ostream& s, const SegmentIndex& index) noexcept;
+std::ostream& operator<<(std::ostream& s,
```
Comment (wthrowe) [inline src/Domain/ElementIndex.hpp]: Should be in namespace domain for ADL.

File: src/Domain/FaceNormal.hpp [hunk 15]
```diff
@@ -43,13 +46,13 @@ class Mesh;
  */
 template <size_t VolumeDim, typename TargetFrame>
 tnsr::i<DataVector, VolumeDim, TargetFrame> unnormalized_face_normal(
-    const Mesh<VolumeDim - 1>& interface_mesh,
+    const domain::Mesh<VolumeDim - 1>& interface_mesh,
     const ElementMap<VolumeDim, TargetFrame>& map,
     const Direction<VolumeDim>& direction) noexcept;
```
Comment (wthrowe) [inline src/Domain/FaceNormal.hpp]: Why does `Mesh` get a namespace but not `ElementMap` or `Direction`?  Similar things happen in other places as well.

File: src/Domain/Neighbors.cpp [hunk 16]
```diff
@@ -47,19 +44,30 @@ void Neighbors<VolumeDim>::pup(PUP::er& p) noexcept {
...
+      const domain::Neighbors<DIM(data)>& lhs,                      \
+      const domain::Neighbors<DIM(data)>& rhs) noexcept;            \
+  template bool domain::operator!=(                                 \
+      const domain::Neighbors<DIM(data)>& lhs,                      \
+      const domain::Neighbors<DIM(data)>& rhs) noexcept;
 
-template bool operator==(const Neighbors<1>&, const Neighbors<1>&) noexcept;
-template bool operator==(const Neighbors<2>&, const Neighbors<2>&) noexcept;
-template bool operator==(const Neighbors<3>&, const Neighbors<3>&) noexcept;
+GENERATE_INSTANTIATIONS(INSTANTIATE, (1, 2, 3))
```
Comment (wthrowe) [inline src/Domain/Neighbors.cpp:65]: Factor out.  (And can't you instantiate inside the namespace and avoid all the qualifiers?)

File: src/Executables/Benchmark/Benchmark.cpp [hunk 17]
```diff
@@ -103,6 +104,7 @@ void bench_all_gradient(benchmark::State& state) {  // NOLINT
 BENCHMARK(bench_all_gradient);
 }  // namespace
 
-BENCHMARK_MAIN()
-
-#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.tpp"
+// Some compilers require a semi-colon here with some versions of Benchmark. If
+// you get "xxpected initialzer at end of input" errors then you need to add the
+// semi-colon.
+BENCHMARK_MAIN()  // ;
```
Comment (wthrowe) [inline src/Executables/Benchmark/Benchmark.cpp]: This change seems unrelated, and "expected" and "initializer" are spelled wrong.

**Review by wthrowe — CHANGES_REQUESTED** [review]
Some of my comments were general and just placed on the specific example I first noticed.  I have not looked at the tests at all.

**Review by kidder — COMMENTED** [review]
Sigh, it would help if I actually submitted reviews when we did them...  :(

File: src/Domain/ElementId.cpp (hunk 1 above)
Comment (nilsdeppe) [reply src/Domain/ElementId.cpp]: ADL lookup means we it will only find the stream operator inside the enclosing namespace. In this case that's `std`, but we aren't allowed to put things into `std` so we need to specify where the operator lives.

File: src/Domain/ElementIndex.cpp (hunk 12 above)
Comment (nilsdeppe) [reply src/Domain/ElementIndex.cpp]: No

File: src/Domain/FaceNormal.hpp (hunk 15 above)
Comment (nilsdeppe) [reply src/Domain/FaceNormal.hpp]: Because it's much easier to namespace one class at a time rather than 30....

File: src/Domain/Neighbors.cpp (hunk 16 above)
Comment (nilsdeppe) [reply src/Domain/Neighbors.cpp:65]: No, and yes

File: src/DataStructures/VariablesHelpers.hpp [hunk 18]
```diff
@@ -13,14 +13,15 @@
 
 #include "DataStructures/DataVector.hpp"
 #include "DataStructures/SliceIterator.hpp"
-#include "DataStructures/Tensor/IndexType.hpp"
```
Comment (kidder) [inline src/DataStructures/VariablesHelpers.hpp]: this should go in one of the IWYU commits

File: src/Domain/BlockNeighbor.cpp [hunk 19]
```diff
@@ -3,10 +3,11 @@
 
 #include "Domain/BlockNeighbor.hpp"
 
-#include "Utilities/GenerateInstantiations.hpp"
```
Comment (kidder) [inline src/Domain/BlockNeighbor.cpp]: all changes to this file except the two lines adding the namespace should be in the Generate Instantiations commit

File: src/Domain/CoordinateMaps/EquatorialCompression.cpp [hunk 20]
```diff
@@ -3,7 +3,7 @@
 
 #include "Domain/CoordinateMaps/EquatorialCompression.hpp"
 
-#include <cmath>
```
Comment (kidder) [inline src/Domain/CoordinateMaps/EquatorialCompression.cpp]: this should be in the IWYU commit

File: src/Domain/DomainHelpers.cpp [hunk 21]
```diff
@@ -509,7 +510,8 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
     wedges_and_half_wedges.push_back(wedges[5]);
 
     // clang-tidy: trivially copyable
-    return make_vector_coordinate_map_base<Frame::Logical, TargetFrame, 3>(
+    return make_vector_coordinate_map_base<Frame::Logical, TargetFrame,
```
Comment (kidder) [inline src/Domain/DomainHelpers.cpp]: random format changes like the several in this file are annoying in a large commit

File: src/Domain/Element.cpp [hunk 22]
```diff
@@ -7,8 +7,10 @@
 #include <pup.h>  // IWYU pragma: keep
 
 #include "Parallel/PupStlCpp11.hpp"  // IWYU pragma: keep
+#include "Utilities/GenerateInstantiations.hpp"
```
Comment (kidder) [inline src/Domain/Element.cpp:10]: Genarate Instantiation changes should be in that commit

File: src/Domain/ElementId.cpp [hunk 23]
```diff
@@ -80,32 +82,31 @@ size_t hash_value(const ElementId<VolumeDim>& c) noexcept {
...
 
-template class ElementId<1>;
-template class ElementId<2>;
-template class ElementId<3>;
+#define DIM(data) BOOST_PP_TUPLE_ELEM(0, data)
 
-template std::ostream& operator<<(std::ostream&, const ElementId<1>&);
-template std::ostream& operator<<(std::ostream&, const ElementId<2>&);
-template std::ostream& operator<<(std::ostream&, const ElementId<3>&);
+#define INSTANTIATE(_, data)                                        \
```
Comment (kidder) [inline src/Domain/ElementId.cpp:99]: I'm willing to let this one stay in this commit since it needed the namespaces within the macro

File: src/Domain/ElementIndex.cpp (hunk 12 above)
Comment (kidder) [reply src/Domain/ElementIndex.cpp]: since the namespace is used within the macro, I am willing to let it stay

File: src/Domain/Direction.cpp [hunk 24]
```diff
@@ -66,9 +66,8 @@ std::ostream& operator<<(std::ostream& os,
 template std::ostream& operator<<(std::ostream&, const Direction<1>&);
 template std::ostream& operator<<(std::ostream&, const Direction<2>&);
 template std::ostream& operator<<(std::ostream&, const Direction<3>&);
-
-/// \cond
 template void Direction<1>::pup(PUP::er&) noexcept;
 template void Direction<2>::pup(PUP::er&) noexcept;
 template void Direction<3>::pup(PUP::er&) noexcept;
 /// \endcond
+}  // namespace domain
```
Comment (kidder) [inline src/Domain/Direction.cpp]: we seem to have putting the \endcond after the end of the namespace if the \cond was before the namespace

File: src/Domain/ElementMap.cpp [hunk 25]
```diff
@@ -71,3 +72,4 @@ template class ElementMap<1, Frame::Grid>;
 template class ElementMap<2, Frame::Grid>;
 template class ElementMap<3, Frame::Grid>;
 /// \endcond
+}  // namespace domain
```
Comment (kidder) [inline src/Domain/ElementMap.cpp]: \endcond after namespace

File: src/Domain/InitialElementIds.cpp [hunk 26]
```diff
@@ -87,3 +89,5 @@ template std::vector<ElementId<2>> initial_element_ids<2>(
 template std::vector<ElementId<3>> initial_element_ids<3>(
     const std::vector<std::array<size_t, 3>>&
         initial_refinement_levels) noexcept;
+/// \endcond
```
Comment (kidder) [inline src/Domain/InitialElementIds.cpp]: endcond after namespace

File: src/Domain/Neighbors.cpp (hunk 16 above)
Comment (kidder) [reply src/Domain/Neighbors.cpp:65]: this one should be factored out

File: src/Domain/Neighbors.hpp [hunk 27]
```diff
@@ -110,3 +113,5 @@ bool operator==(const Neighbors<VolumeDim>& lhs,
 template <size_t VolumeDim>
 bool operator!=(const Neighbors<VolumeDim>& lhs,
                 const Neighbors<VolumeDim>& rhs) noexcept;
+
```
Comment (kidder) [inline src/Domain/Neighbors.hpp:116]: remove blank line

File: src/Domain/OrientationMap.hpp [hunk 28]
```diff
@@ -7,18 +7,17 @@
 #include <cstddef>
 #include <iosfwd>
 
-
 #include "Domain/Direction.hpp"
-#include "Domain/SegmentId.hpp"
+#include "Domain/SegmentId.hpp"  // IWYU pragma: keep
```
Comment (kidder) [inline src/Domain/OrientationMap.hpp:11]: move into IWYU commit

File: src/Domain/Tags.hpp [hunk 29]
```diff
@@ -40,19 +40,20 @@ namespace OptionTags {
...
 namespace Tags {
 /// \ingroup DataBoxTagsGroup
 /// \ingroup ComputationalDomainGroup
-/// The ::Element associated with the DataBox
+/// The domain::Element associated with the DataBox
 template <size_t VolumeDim>
 struct Element : db::SimpleTag {
   static std::string name() noexcept { return "Element"; }
-  using type = ::Element<VolumeDim>;
+  using type = ::domain::Element<VolumeDim>;
```
Comment (kidder) [inline src/Domain/Tags.hpp]: why does this one have the `::` before the domain, but the others don't?

File: src/Domain/Tags.hpp [hunk 30]
```diff
@@ -263,7 +267,8 @@ struct evaluate_compute_item<DirectionsTag, BaseComputeItem,
   using volume_tags = typename volume_tags<BaseComputeItem>::type;
   static_assert(
       tmpl::size<tmpl::list_difference<
-          volume_tags, typename BaseComputeItem::argument_tags>>::value == 0,
```
Comment (kidder) [inline src/Domain/Tags.hpp]: more random format changes

File: src/NumericalAlgorithms/Interpolation/IrregularInterpolant.hpp [hunk 31]
```diff
@@ -5,7 +5,6 @@
 
 #include <cstddef>
 
-#include "DataStructures/DataVector.hpp"
```
Comment (kidder) [inline src/NumericalAlgorithms/Interpolation/IrregularInterpolant.hpp:8]: another IWYU change

**Review by kidder — COMMENTED** [review]
Made it through src still need to go through tests

**Comment by nilsdeppe** [comment]
Things that are not changing:
- Where IWYU pragmas are. We often have them in other commits as collateral. They are trivial to review.
- The generate instantiations. We've always taken the "If it links it works" approach.
- We've allowed a lot of people, including you, to have blank lines before the closing namespace brace.

Re the formatting comments: I've dealt with the ones you pointed. However, I think we should reconsider enforcing clang-format. Having to judge whether formatting is correct as a human is difficult and will become increasingly difficult as more people contribute. It's a machine checkable thing, so it should be machine checked.

File: tests/Unit/Domain/CoordinateMaps/Test_Translation.cpp [hunk 32]
```diff
@@ -19,7 +19,7 @@
 #include "tests/Unit/Domain/CoordinateMaps/TestMapHelpers.hpp"
 #include "tests/Unit/TestHelpers.hpp"
 
-SPECTRE_TEST_CASE("Unit.Domain.CoordMapsTimeDependent.Translation",
+SPECTRE_TEST_CASE("Unit.Domain.domain::CoordMapsTimeDependent.Translation",
```
Comment (kidder) [inline tests/Unit/Domain/CoordinateMaps/Test_Translation.cpp]: fix this

File: tests/Unit/Domain/DomainTestHelpers.cpp [hunk 33]
```diff
@@ -181,61 +185,67 @@ bool blocks_are_neighbors(
...
+domain::OrientationMap<VolumeDim> find_neighbor_orientation(
+    const domain::Block<VolumeDim, TargetFrame>& host_block,
+    const domain::Block<VolumeDim, TargetFrame>& neighbor_block) noexcept {
   for (const auto& neighbor : host_block.neighbors()) {
     if (neighbor.second.id() == neighbor_block.id()) {
       return neighbor.second.orientation();
     }
   }
-  ERROR("The Block `neighbor_block` is not a neighbor of `host_block`.");
+  ERROR(
```
Comment (kidder) [inline tests/Unit/Domain/DomainTestHelpers.cpp]: do not put the namespace in the error message

File: tests/Unit/Domain/DomainTestHelpers.cpp [hunk 34]
```diff
@@ -181,61 +185,67 @@ bool blocks_are_neighbors(
...
+domain::Direction<VolumeDim> find_direction_to_neighbor(
+    const domain::Block<VolumeDim, TargetFrame>& host_block,
+    const domain::Block<VolumeDim, TargetFrame>& neighbor_block) noexcept {
   for (const auto& neighbor : host_block.neighbors()) {
     if (neighbor.second.id() == neighbor_block.id()) {
       return neighbor.first;
     }
   }
-  ERROR("The Block `neighbor_block` is not a neighbor of `host_block`.");
+  ERROR(
```
Comment (kidder) [inline tests/Unit/Domain/DomainTestHelpers.cpp]: again

File: tests/Unit/Domain/Test_Mesh.cpp [hunk 35]
```diff
@@ -38,66 +38,65 @@ void test_extents_basis_and_quadrature(
...
-                             Spectral::Quadrature::GaussLobatto};
-    test_extents_basis_and_quadrature(
-        mesh3d_lgl, {{3, 3, 3}},
-        {{Spectral::Basis::Legendre, Spectral::Basis::Legendre,
-          Spectral::Basis::Legendre}},
-        {{Spectral::Quadrature::GaussLobatto,
-          Spectral::Quadrature::GaussLobatto,
-          Spectral::Quadrature::GaussLobatto}});
+SPECTRE_TEST_CASE("Unit.Domain.Mesh", "[Domain][Unit]"){
+    SECTION("Uniform LGL mesh"){const domain::Mesh<1> mesh1d_lgl{
```
Comment (kidder) [inline tests/Unit/Domain/Test_Mesh.cpp]: fix this

**Review by kidder — CHANGES_REQUESTED** [review]
Also, your namespace is undocumented and not in the computational domain group

**Comment by kidder** [comment]
only clang-tidy failure is from boost

**Review by kidder — DISMISSED** [review]

File: tests/Unit/Domain/DomainTestHelpers.cpp [hunk 36]
```diff
@@ -1,10 +1,10 @@
 // Distributed under the MIT License.
 // See LICENSE.txt for details.
 
-#include "tests/Unit/Domain/DomainTestHelpers.hpp"
-
 #include "tests/Unit/TestingFramework.hpp"
 
+#include "tests/Unit/Domain/DomainTestHelpers.hpp"
+
```
Comment (fmahebert) [inline tests/Unit/Domain/DomainTestHelpers.cpp:7]: Can you also update the [code review guide](https://spectre-code.org/code_review_guide.html), which currently does not give any indication that only one of these orderings is correct?

I suggest replacing what are currently lines 1 and 2 with the following text:
> 1. (if a test:) tests/Unit/TestingFramework.hpp, followed by a blank line
> 2. (if a cpp with a corresponding hpp file:) hpp corresponding to cpp, followed by a blank line

Comment (nilsdeppe) [reply tests/Unit/Domain/DomainTestHelpers.cpp:7]: No, we need to move on.

Comment (fmahebert) [reply tests/Unit/Domain/DomainTestHelpers.cpp:7]: Okay, then I'll make this correction in a "various small fixes" PR I'm preparing. I do feel that our code review guide should document our requirements.

Comment (nilsdeppe) [reply tests/Unit/Domain/DomainTestHelpers.cpp:7]: Sounds good!

**Comment by nilsdeppe** [comment]
Rebased on develop after #784 was merged

File: src/Domain/Tags.hpp [hunk 37]
```diff
@@ -40,19 +40,20 @@ namespace OptionTags {
...
 struct DomainCreator {
-  using type = std::unique_ptr<::DomainCreator<Dim, TargetFrame>>;
+  using type = std::unique_ptr<domain::DomainCreator<Dim, TargetFrame>>;
   static constexpr OptionString help = {"The domain to create initially"};
 };
 }  // namespace OptionTags
 
+namespace domain {
 namespace Tags {
 /// \ingroup DataBoxTagsGroup
```
Comment (nilsvu) [inline src/Domain/Tags.hpp:50]: I think with the `Tags` namespace now enclosed in `domain` you should document it as something like `\ref DataBoxGroup tags associated with the computational domain` to make it show up in Doxygen. Then you can also remove the `\ingroup DataBoxTagsGroup` from the tags below, since they are nicely listed in the namespace.

**Comment by nilsdeppe** [comment]
rebased on develop

**Review by kidder — APPROVED** [review]

File: tests/Unit/Domain/DomainTestHelpers.cpp (hunk 36 above)
Comment (fmahebert) [reply tests/Unit/Domain/DomainTestHelpers.cpp:7]: This is included in PR #964

**Comment by kidder** [comment]
Current status:

- [ ] @nilsdeppe has broken this into smaller PRs, some of which have been merged.  
- [ ] @nilsdeppe will close this PR when all the smaller PRs have been merged

**Comment by kidder** [comment]
closing this as it is out of date and partially implemented.  @nilsdeppe agrees it will be easier to just open a new PR to finish adding the namespace
